libdnf/repo/temp_files_memory.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "temp_files_memory.hpp"
+
+#include "utils/bgettext/bgettext-mark-domain.h"
+#include "utils/fs/file.hpp"
+
+#include "libdnf/common/exception.hpp"
+
+#include <toml.hpp>
+
+
+namespace libdnf::repo {
+
+TempFilesMemory::TempFilesMemory(const std::string & parent_dir)
+    : full_memory_path(std::filesystem::path(parent_dir) / MEMORY_FILENAME) {}
+
+TempFilesMemory::~TempFilesMemory() = default;
+
+std::vector<std::string> TempFilesMemory::get_files() const {
+    if (!std::filesystem::exists(full_memory_path)) {
+        return {};
+    }
+
+    try {
+        auto toml_data = toml::parse(full_memory_path);
+        return toml::get<std::vector<std::string>>(toml_data[FILE_PATHS_TOML_KEY]);
+    } catch (const toml::exception & e) {
+        throw libdnf::Error(
+            M_("An error occurred when parsing the temporary files memory file at '{}': {}"),
+            full_memory_path.string(),
+            std::string(e.what()));
+    }
+}
+
+void TempFilesMemory::add_files(const std::vector<std::string> & paths) {
+    auto files = get_files();
+    files.insert(files.end(), paths.begin(), paths.end());
+
+    // sort and deduplicate resulting vector
+    std::sort(files.begin(), files.end());
+    files.erase(std::unique(files.begin(), files.end()), files.end());
+
+    utils::fs::File(full_memory_path, "w").write(toml::format(toml::value({{FILE_PATHS_TOML_KEY, files}})));
```

Comment:
I guess that the line could generate various number of errors. Is it covered somehow. 
Is the operation safe when multiple instances of DNF will be modifying the file?

Reply:
> I guess that the line could generate various number of errors. Is it covered somehow.

It is OK, the exception part is well described in method. Perfect job.

Reply:
I think you are not referring to the latest changes in the file which is weird as the PR is already updated for some time... Anyway, this should be handled by this commit: https://github.com/rpm-software-management/dnf5/pull/459/commits/709f245cfa2189bd06af0b3bb1771f19e475e9f2.

Reply:
I am sorry, I just realized that in the next commit